### PR TITLE
HPCC-31938 Fix issues with PERSIST queue switching in k8s

### PR DESCRIPTION
**Pull request by jakesmith**
## Type of change:
- [x] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Cloud-compatibility
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

**Review by ghalliday — COMMENTED** [review]
@jakesmith looks good.  A couple of questions, but could merge as-is.

File: common/workunit/workunit.cpp [hunk 1]
```diff
@@ -14313,17 +14314,36 @@ void executeThorGraph(const char * graphName, IConstWorkUnit &workunit, const IP
...
+                queue = tgt;
+            }
+        }
 
         // NB: executeGraphOnLingeringThor looks for existing Thor instance that has been used for the same job,
         // and communicates with it directly
         if (!multiJobLinger && executeGraphOnLingeringThor(workunit, wfid, graphName))
+        {
+            if (!thisThor)
+                throw makeStringExceptionV(0, "multiJobLinger mode required to targeting other thor instances. Target: %s", tgt);
```
[the remark this answers is not in the document]
Comment (jakesmith) [reply common/workunit/workunit.cpp:14337]: will fix

File: helm/hpcc/templates/thor.yaml [hunk 2]
```diff
@@ -28,10 +28,10 @@ Pass in dict with root and me
 {{- define "hpcc.thorConfigMap" -}}
 {{- $eclAgentType := .me.eclAgentType | default "hthor" }}
 {{- $hthorName := printf "%s-%s" .me.name $eclAgentType }}
-{{- $eclAgentScope := dict "name" .eclAgentName "type" $eclAgentType "useChildProcesses" .eclAgentUseChildProcesses "replicas" .eclAgentReplicas "maxActive" .me.maxJobs | merge (pick .me "keepJobs" "logging" "tracing" "auxQueues") }}
-{{- $thorAgentScope := dict "name" .thorAgentName "replicas" .thorAgentReplicas "maxActive" .me.maxGraphs | merge (pick .me "keepJobs" "logging" "tracing") }}
+{{- $eclAgentScope := dict "name" .eclAgentName "type" $eclAgentType "useChildProcesses" .eclAgentUseChildProcesses "replicas" .eclAgentReplicas "maxActive" .me.maxJobs | merge (pick .me "keepJobs" "logging" "tracing" "auxQueues") ((hasKey .me "holdAgent") | ternary (dict "hold" .me.holdAgent) dict) }}
```
[the remark this answers is not in the document]
Comment (jakesmith) [reply helm/hpcc/templates/thor.yaml:31]: should be in separate PR really.. had meant to tbh.
but if you're okay with leaving it in this one ..

I'll add a some commented out examples to values yaml

**Review by jakesmith — COMMENTED** [review]
@ghalliday - please see new commit.

**Review by ghalliday — APPROVED** [review]
Please squash and I will merge

**Pull request merged**